### PR TITLE
[Bug Fix] Allow logger to support indexing

### LINKED ISSUE #4540: Existing logger conflict with lr_finder
## 🐛 Bug

If you call specific logger function (ie. `add_histogram` on a tensorboard one). The tuner will fails because it overwrite it with a DummyLogger (https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/tuner/lr_finder.py#L150).
I would expect to not have to check if the logger I initiated in my trainer is being overwritten or not.
Additionally, on a minor minor note. I wouldn't expect `progress_bar_refresh_rate` to be overwritten as it is [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/tuner/lr_finder.py#L147). 

### Expected behavior

Have a smooth way to ignore it. 

### Environment

```
* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.18.5
	- pyTorch_debug:     True
	- pyTorch_version:   1.7.0+cu101
	- pytorch-lightning: 0.10.0
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.6.9
	- version:           #1 SMP Thu Jul 23 08:00:38 PDT 2020
```

 - PyTorch Version (e.g., 1.0): 1.0
 - OS (e.g., Linux): colab
 - How you installed PyTorch (`conda`, `pip`, source): pip

### Additional context

I am putting this as a bug because it is a behavior that required me to check into the source code to figure out where this DummyLogger came from. If the solution is non-trivial, a resolution could just be some warnings in the doc.

### PR DESCRIPTION
**Pull request by zippeurfou**
## What does this PR do?

This should fix  #4540

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- na Did you make sure to update the documentation with your changes?
- na Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- na If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In in short, see following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [X] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by SkafteNicki** [comment]
Hi @zippeurfou 
Please include a test that would currently fail on master but pass with this PR :]

**Comment by zippeurfou** [comment]
Sure happy to do so.

**Comment by Borda** [comment]
@SkafteNicki mind bring a bit light on how this fixes the LF issue?
eventually, this yields an infinite loop if you have miss `LoggerCollection` and base logger

**Review by Borda — COMMENTED** [review]
@SkafteNicki mind bring a bit light on how this fixes the LF issue?
eventually, this yields an infinite loop if you have miss `LoggerCollection` and base logger

**Comment by SkafteNicki** [comment]
@Borda the problem currently is that neither `DummyExperiment` and `DummyLogger` supports indexing, so if the users has something like this in their code:
`self.logger[0].experiment.add_image(...)`
or
`self.logger.experiment[0].add_image(...)`
currently lr finder will crash as it temporarily replaces the user logger with a `DummyLogger`.

Can you elaborate in which case this can give an infinite loop?

File: pytorch_lightning/loggers/base.py [hunk 1]
```diff
@@ -386,6 +386,9 @@ def nop(*args, **kw):
     def __getattr__(self, _):
         return self.nop
 
+    def __getitem__(self, idx):
+        return self
```
Comment (tchaton) [inline pytorch_lightning/loggers/base.py:390]:
```suggestion
    def __getitem__(self, idx):
        # enables self.logger[0].experiment.add_image 
        # and self.logger.experiment[0].add_image(...)
        return self
```

**Review by tchaton — COMMENTED** [review]
Looks slightly weird to me. Do you have any other solution in mind ?

**Review by awaelchli — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by Borda — APPROVED** [review]
lgtm

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**